libs/templates/src/lib/trade/table/trading-spec-table.tsx

```diff
@@ -0,0 +1,306 @@
+import { useEffect, useState } from 'react';
+import { mainInfo, trading_condition, additional_info } from '../data/data';
+import { TableDataType } from '../types/types';
+import {
+  flexRender,
+  getCoreRowModel,
+  getSortedRowModel,
+  useReactTable,
+  getPaginationRowModel,
+} from '@tanstack/react-table';
+import UseColumns from '../hooks/use-columns';
+import { Button, qtMerge, Section, Text } from '@deriv/quill-design';
+import {
+  LabelPairedEllipsisVerticalBoldIcon,
+  StandaloneChevronLeftRegularIcon,
+  StandaloneChevronRightRegularIcon,
+  StandaloneCircleDotFillIcon,
+  StandaloneCircleRegularIcon,
+  StandaloneXmarkRegularIcon,
+} from '@deriv/quill-icons';
+import { SearchChip } from '@deriv-com/components';
+import { BottomSheet } from '@deriv-com/components';
+
+const TradingSpecTable = () => {
+  const [data, setData] = useState<TableDataType[]>([]);
+  const [selected_filter, setSelectedFilter] = useState<TableDataType[]>(
+    mainInfo.data,
+  );
+  const [selectedInfo, setSelectedInfo] = useState('mainInfo');
+  const [searchValue, setSearchValue] = useState('');
+  const [showBottomSheet, setShowBottomSheet] = useState(false);
+
+  const handleSubmit = (e: React.ChangeEvent<HTMLInputElement>) => {
+    e.preventDefault();
+  };
+
+  useEffect(() => {
+    let updatedRowData = [];
+    if (searchValue.length >= 1) {
+      updatedRowData = [
+        ...selected_filter.filter(
+          (item) =>
+            item.instrument?.instrument?.match(new RegExp(searchValue, 'i')),
+        ),
+      ];
+
+      setData(updatedRowData);
+    } else {
+      setData(selected_filter);
+    }
+  }, [searchValue, selected_filter]);
+
+  const columns = UseColumns(selectedInfo);
+
+  const table = useReactTable({
+    data,
+    columns,
+    getCoreRowModel: getCoreRowModel(),
+    getSortedRowModel: getSortedRowModel(),
+    getPaginationRowModel: getPaginationRowModel(),
+  });
+  useEffect(() => {
+    table.setPageSize(15);
+  }, [table]);
+  useEffect(() => {
+    document.body.style.overflow = showBottomSheet ? 'hidden' : 'scroll';
+  }, [showBottomSheet]);
+  return (
+    <Section
```

Comment:
can you please break this down to smaller components? 
this is gonna be really hard to maintain
thanks thanks 🙏

Reply:
hi @mohsen-deriv chips and pagination will be done from quill design. i have used buttons with conditions here, that's why it looks huge.